src/rum/instrumentation/DdRumResourceTracking.tsx

```diff
@@ -5,14 +5,66 @@
  */
 
 import { DdRum } from '../../foundation'
-import type { DdRumXhr } from './DdRumXhr'
+import type { DdRumXhr, DdRumXhrContext } from './DdRumXhr'
 import { generateTraceId } from './TraceIdentifier';
 
 export const TRACE_ID_HEADER_KEY = "x-datadog-trace-id"
 export const PARENT_ID_HEADER_KEY = "x-datadog-parent-id"
 export const ORIGIN_HEADER_KEY = "x-datadog-origin"
 export const ORIGIN_RUM = "rum"
 
+const MISSING_TIME = -1
+
+type Duration = number & { d: 'Duration in ms' }
```

Comment:
can you explain this syntax ? It looks weird to me but probably is TS normal syntax.

Reply:
this is type intersection https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html. I think I will just remove it.